notebooks/06a Modelling - classification.ipynb

```diff
@@ -0,0 +1,865 @@
+    "1 - Very low risk|0-6\n",
+    "2 - Low risk|7-10\n",
+    "3 - Normal risk|11-13\n",
+    "4 - Elevated risk|14-15\n",
+    "5 - High risk|>15"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "gather": {
+     "logged": 1653046712712
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# actual risk scores\n",
+    "risk_labels = [\n",
+    "    \"1 - Very Low Risk\",\n",
+    "    \"2 - Low Risk\",\n",
+    "    \"3 - Normal Risk\",\n",
+    "    \"4 - Elevated Risk\",\n",
+    "    \"5 - High Risk\",\n",
+    "]\n",
+    "features_df[\"risk\"] = [risk_score(los) for los in features_df.LENGTH_OF_STAY]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Define target and training features"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "gather": {
+     "logged": 1653046712865
+    }
+   },
+   "outputs": [],
+   "source": [
+    "X = features_df.drop(columns=[\"LENGTH_OF_STAY\"])\n",
+    "y = features_df.risk"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Train/test split\n",
+    "\n",
+    "For model evaluation, we will hold back a 25% test set, and use cross-validation on the remaining 75% for all models until the final comparison is made."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "gather": {
+     "logged": 1653046713027
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# Split data for train/test\n",
+    "X_train, X_test, y_train, y_test = train_test_split(\n",
+    "    X, y, train_size=0.75, random_state=42\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Explore class imbalance\n",
+    "\n",
+    "This is a multi-class classification model, so we need to understand where any class imbalance lies otherwise we will skew to larger classes."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "gather": {
+     "logged": 1653046713201
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# Show how many of each class are present in the training set:\n",
+    "X_train.risk.value_counts().sort_index()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Weight samples\n",
+    "\n",
+    "Given significant class imbalance, we will weight samples where the smallest class has a weight of 1, and other classes as a (<1) proportion.\n",
+    "\n",
+    "e.g.\n",
+    "\n",
+    "class|count|weight\n",
+    "---|---|---\n",
+    "1|1000|0.1\n",
+    "2|100|1\n",
+    "3|500|0.2"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "gather": {
+     "logged": 1653046713351
+    }
+   },
+   "outputs": [],
+   "source": [
+    "weights = X_train.risk.value_counts().min() / X_train.risk.value_counts().sort_index()\n",
```

Comment:
Up to you but the models you use all have a class_weight argument where 'balanced' is an option which will do this for you.